lexos/static/js/scripts_scrub.js

```diff
@@ -1,267 +1,309 @@
+    response = JSON.parse(response)
+    $('#preview-body').empty()
+    $.each(response['data'], function () {
+      const fileName = $(this)[1]
+      const fileContents = $(this)[3]
+      const fieldSet = $('<fieldSet></fieldSet>')
+      // CSS truncates the document name
+      fieldSet.append('<legend class="has-tooltip" style="color:#999; width:90%;margin: auto; white-space: nowrap; overflow: hidden; text-overflow: ellipsis;">' + fileName + '</legend>')
+      fieldSet.append('<div class="filecontents">' + fileContents + '</div>') // Keep this with no whitespace!
+      $('#preview-body').append(fieldSet)
+      $('#status-prepare').css({'visibility': 'hidden'})
+    })
   })
+}
+
+/**
+ * Function to handle the chevron drop down button rotate animation by toggling
+ * the class so the appropriate CSS applies.
+ * @returns {void} - returns nothing
+ */
+function rotateChevron () {
+  $(this).find('span').toggleClass('down')
+  $(this).next().collapse('toggle')
+}
+
+/**
+ * Get additional options and truncate file names when necessary
+ * @returns {void} - returns nothing
+ */
+function displayAdditionalOptions () {
+  const advancedOptions = $('#advanced-title')
+  advancedOptions.find('.icon-arrow-right').addClass('showing')
+  advancedOptions.siblings('.expansion').slideToggle(0)
+
+  $('#swfileselect').change(truncateFileName())
+
+  $('#lemfileselect').change(truncateFileName())
+
+  $('#consfileselect').change(truncateFileName())
+
+  $('#scfileselect').change(truncateFileName())
+}
 
-  $('#punctbox').mousedown(function () {
-    var timeToToggle = 300
+/**
+ * Truncate file names if needed
+ * @param ev - event? not sure how to pass in function call...
+ * @returns {void}
+ */
+function truncateFileName (ev) {
+  let fileName = ev.target.files[0].name
+  if (fileName.length > 25) { fileName = fileName.substring(0, 24) + '...' }
+  $(this).html(fileName)
+}
+
+/**
+ * Clone file labels and do ajax request
+ * @returns {void} - returns nothing
+ */
+function buttonFileLabelsFunction () {
+  // swfileselect, lemfileselect, consfileselect, scfileselect
+  const fileType = $(this).attr('id').replace('bttnlabel', '')
+  const usingCache = $('#usecache' + fileType).attr('disabled') !== 'disabled'
+
+  if ((usingCache) || ($(this).attr('id') !== '')) {
+    // $(this).siblings('.scrub-upload').attr('value', '');
+    // Next two lines clear the file input; it's hard to find a cross-browser solution
+    $('#' + fileType).val('')
+    $('#' + fileType).replaceWith($('#' + fileType).clone(true))
+    $('#usecache' + fileType).attr('disabled', 'disabled')
+    $(this).text('')
+  }
 
-    if ($('#aposhyph')[0].style.cssText == 'display: none;') {
-      $('#aposhyph').fadeIn(timeToToggle)
-    } else {
-      $('#aposhyph').fadeOut(timeToToggle)
-      $('#aposhyph')[0].style.cssText == 'display: none;'
+  // Do Ajax
+  $.ajax({
+    type: 'POST',
+    url: '/removeUploadLabels',
+    data: $(this).text().toString(),
```

Comment:
I still don't know what is `this`...

Reply:
I am actually not sure what `this` is in this context. The script in the HTML is poorly documented and its purpose is not immediately apparent.

Reply:
I do not believe this ajax call is necessary. It does not do anything with the response. I just removed it and tested and it seems to have no impact.

Reply:
I think this may be some bad practice on the backend. Please investigate on what /removeUploadLabels page does.